src/interface/obsidian/src/chat_view.ts

```diff
@@ -0,0 +1,1306 @@
+            this.renderMessage(chatBodyEl, errorMsg, "khoj", undefined);
+            return false;
+        }
+        return true;
+    }
+
+    async readChatStream(response: Response, responseElement: HTMLDivElement): Promise<void> {
+        // Exit if response body is empty
+        if (response.body == null) return;
+
+        const reader = response.body.getReader();
+        const decoder = new TextDecoder();
+
+        while (true) {
+            const { value, done } = await reader.read();
+
+            // Break if the stream is done
+            if (done) break;
+
+            let responseText = decoder.decode(value);
+            if (responseText.includes("### compiled references:")) {
+                // Render any references used to generate the response
+                const [additionalResponse, rawReference] = responseText.split("### compiled references:", 2);
+                await this.renderIncrementalMessage(responseElement, additionalResponse);
+
+                const rawReferenceAsJson = JSON.parse(rawReference);
+                let references = this.extractReferences(rawReferenceAsJson);
+                responseElement.appendChild(this.createReferenceSection(references));
+            } else {
+                // Render incremental chat response
+                await this.renderIncrementalMessage(responseElement, responseText);
+            }
+        }
+    }
+
+    async getChatResponse(query: string | undefined | null): Promise<void> {
+        // Exit if query is empty
+        if (!query || query === "") return;
+
+        // Render user query as chat message
+        let chatBodyEl = this.contentEl.getElementsByClassName("khoj-chat-body")[0] as HTMLElement;
+        this.renderMessage(chatBodyEl, query, "you");
+
+        let conversationID = chatBodyEl.dataset.conversationId;
+        if (!conversationID) {
+            let chatUrl = `${this.setting.khojUrl}/api/chat/sessions?client=obsidian`;
+            let response = await fetch(chatUrl, {
+                method: "POST",
+                headers: { "Authorization": `Bearer ${this.setting.khojApiKey}` },
+            });
+            let data = await response.json();
+            conversationID = data.conversation_id;
+            chatBodyEl.dataset.conversationId = conversationID;
+        }
+
+        // Get chat response from Khoj backend
+        let encodedQuery = encodeURIComponent(query);
+        let chatUrl = `${this.setting.khojUrl}/api/chat?q=${encodedQuery}&n=${this.setting.resultsCount}&client=obsidian&stream=true&region=${this.location.region}&city=${this.location.city}&country=${this.location.countryName}&timezone=${this.location.timezone}`;
+        let responseElement = this.createKhojResponseDiv();
+
+        // Temporary status message to indicate that Khoj is thinking
+        this.result = "";
+        let loadingEllipsis = this.createLoadingEllipse();
```

Comment:
the loading ellipse doesn't seem to be showing up for me, but not sure why.

Reply:
Yeah, the loading ellipsis needed to be of a contrasting color. Fixed that